crates/polars-core/src/frame/horizontal.rs

```diff
@@ -76,7 +76,16 @@ impl DataFrame {
 }
 /// Concat [`DataFrame`]s horizontally.
 /// Concat horizontally and extend with null values if lengths don't match
-pub fn concat_df_horizontal(dfs: &[DataFrame], check_duplicates: bool) -> PolarsResult<DataFrame> {
+pub fn concat_df_horizontal<T>(
+    dfs: &[DataFrame],
+    check_duplicates: bool,
+    strict: T,
```

Comment:
Why the generics here? This compiles fine using a bool.

Reply:
I thought it had to be an optional parameter for Rust as well (not just Python).
In v2 milestone, it would be easier to just change the default value in the function body.
Also if strict is a bool, what should be the value for `strict` in other calls to concat_df_horizontal (ex: zip.rs)?

Reply:
`None` gets converted to false anyway, so I think just passing `false` is the way to go. It makes sense as an optional parameter for a generic concat function (where the parameter is only applicable to in the horizontal case) but in any horizontal-specific function and on the rust side it shouldn't be optional.

Reply:
Understood. Thank you